pkgs/development/python-modules/sourcery/default.nix

```diff
@@ -0,0 +1,30 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub,
+  lib,
+  setuptools,
+}:
+
+buildPythonPackage rec {
+  pname = "sourcery";
+  version = "1.2.0";
+  format = "pyproject";
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  src = fetchFromGitHub {
+    owner = "sourcery-ai";
+    repo = "sourcery";
+    rev = "v${version}";
+    hash = "sha256-KQLTOCMLczUwO6E3uZCdTUfvnr1D3sl4Bw6lWjPXkh8=";
+  };
+
```

Comment:
add an imports check

Reply:
When I add:

```python
  pythonImportsCheck = [ "sourcery-precommit" ];
```

I get:  `ModuleNotFoundError: No module named 'sourcery-precommit'`

Reply:
Indeed, I just noticed.

This PR has been inspired by the work of @superherointj at: https://github.com/superherointj/nix-auto-gpt/tree/master

Should I get rid of this derivation then?

Reply:
If nothing depends on it, and you didn't grab the wrong package, sure.

Reply:
It is listed here: https://github.com/superherointj/nix-auto-gpt/blob/master/shell.nix#L45, I will make some tests and see if ChatGPT works without it. I guess yes.

Reply:
`sourcery` was removed from Auto-GPT. Can ignore it.
https://github.com/Significant-Gravitas/Auto-GPT/commit/bb2066df044b301a7d7534faa755888aa4674608

Reply:
Thanks, removed.